src/commands/lint/args.rs

```diff
@@ -0,0 +1,110 @@
+use clap::{Args, Subcommand};
+
+use crate::{
+    config::CliConfig,
+    external_cli::{
+        arg_builder::ArgBuilder,
+        cargo::{check::CargoCheckArgs, install::AutoInstall},
+    },
+};
+
+#[derive(Debug, Args)]
+pub struct LintArgs {
+    #[clap(subcommand)]
+    pub subcommand: Option<LintSubcommands>,
+    /// Confirm all prompts automatically.
+    #[arg(long = "yes", default_value_t = false)]
+    pub confirm_prompts: bool,
+
+    /// Arguments to forward to `cargo check`.
+    #[clap(flatten)]
+    pub cargo_args: CargoCheckArgs,
+}
+
+impl LintArgs {
+    /// Whether to automatically install missing dependencies.
+    // Only needed with the `rustup` feature
+    #[cfg(feature = "rustup")]
+    pub(crate) fn auto_install(&self) -> AutoInstall {
+        if self.confirm_prompts {
+            AutoInstall::Always
+        } else {
+            AutoInstall::AskUser
+        }
+    }
+
+    #[cfg(not(feature = "rustup"))]
+    pub(crate) fn auto_install(&self) -> AutoInstall {
+        AutoInstall::Never
+    }
```

Comment:
```suggestion
    pub(crate) fn auto_install(&self) -> AutoInstall {
        if cfg!(feature = "rustup") {
            if self.confirm_prompts {
                AutoInstall::Always
            } else {
                AutoInstall::AskUser
            }
        } else {
            // We cannot auto-install `bevy_lint` without Rustup.
            AutoInstall::Never
        }
    }
```

(Nit) This merges the two feature-gated variants into one so that the compiler can check both at once! It's also a bit easier to read